### ISSUE TITLE
Invoke-IcingaCheckUpdates - permission problems are not reflected in plugin output

### ISSUE THREAD

**Issue by K0nne** [Icinga/icinga-powershell-plugins#49]
Hello,

Invoke-IcingaCheckUpdates shows errors in the plugin output, but returns an [OK]:

```
New-Object: Creating an instance of the COM component with CLSID {4CB43D7F-7EEE-4906-8698-60DA1C38F2FE} from the IClassFactory failed due to the following error: 80070005 Access is denied. 
(Exception from HRESULT: 0x80070005 (E_ACCESSDENIED)).InvalidOperation: You cannot call a method on a null-valued expression.
```
**[OK] Check package "Windows Updates"**

I am using the master packages from yesterday.

**Comment by LordHepipud** [comment]
Thank you for the report. I created a PR for this issue [#61].
Could you please test this and verify if it now returns properly an `Unknown` check result with an exception message?

To test this you will require to update the Framework from the current, latest master.

I will add this bugfix to v.1.2.0 then.

Thank you in advance!

**Comment by aheinhold** [comment]
Hi @LordHepipud ,

i tested the plugin with the current master-branch.
The "permission problem" exists only at remote execution. How can i fix this?

**Icinga-Plugin-Output:**
```
Plugin-Ausgabe
[91mNew-Object: [91mCreating an instance of the COM component with CLSID {ID} from the IClassFactory failed due to the following error: 80070005 Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED)).[0m
[91mInvalidOperation: [91mYou cannot call a method on a null-valued expression.[0m
[OK] Check package "Windows Updates"
```

**Remote Commandline Output (Linux-Satellite):**
```
>> ./check_by_pwsh.ps1 "user" 'password' "hostname.fqdn" "Use-Icinga; return Invoke-IcingaCheckUpdates  -Verbosity 2;"
New-Object: Creating an instance of the COM component with CLSID {ID} from the IClassFactory failed due to the following error: 80070005 Access is denied. (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED)).
InvalidOperation: You cannot call a method on a null-valued expression.
[OK] Check package "Windows Updates" (Match All)
\_ [OK] Check package "Pending Update Count" (Match All)
   \_ [OK] Pending Update Count: 0
| 'pending_update_count'=0;;
```

**Local Commandline Output (Windows-System):**
```
icinga> Invoke-IcingaCheckUpdates -verbosity 2
[OK] Check package "Windows Updates" (Match All)
\_ [OK] Check package "Pending Update Count" (Match All)
   \_ [OK] Pending Update Count: 0
| 'pending_update_count'=0;;
0
```

**Comment by LordHepipud** [comment]
Hi @aheinhold,

the fix is currently not inside the master branch yet, but still within the assigned PR because I was unable to determine on how to setup the permissions.
By now I found that it might help to add the user which is executing the checks to the group

```
Distributed COM-User
```

Would you be able to check and validate it adding the user to this group resolves the issue?

**Comment by LordHepipud** [comment]
Hello,

did you have the chance to my suggestion so far?

**Comment by LordHepipud** [comment]
I have setup a new test environment for this and after a while I was able to reproduce this issue properly and investigated further.

The short version: The current plugin does not support `PowerShell Remote Execution`.

The long version:
To be able to fetch Windows updates, we are requiring to access `COM` objects, to query the corresponding data. However, it is not possible to query these information over network connections (like PowerShell Remote). It seems to be possible to connect to a remote system and fetch data over a request, but not while logging in with PowerShell remote and executing these commands.

This is discussed in many topcs, like this one: https://stackoverflow.com/questions/7078958/powershell-remote-microsoft-update-session-access-denied-0x80070005


As comparison I checked the old Windows Update plugin shipped by Icinga 2 and there is the same issue - it does not return pending updates.

As there is no way to grant permissions to query these information, we will have to user another method.

It is suggested to add a Scheduled Task for this action and query the result of this task after triggering the execution. By doing so, the user the task is assigned too will run the Windows Update check and we simply collect the result.

I will move this issue to 1.4.0 as this will require more research and testing.
